app/react/Favorites/components/FavoriteBanner.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { Component } from 'react';
+import { isClient } from 'app/utils';
+
+export type FavoriteBannerProps = {
+  sharedId: string;
+};
+
+const getUwaziFavorites = () =>
+  localStorage.getItem('uwaziFavorites') ? localStorage.getItem('uwaziFavorites')!.split(',') : [];
+
+class FavoriteBanner extends Component<FavoriteBannerProps> {
+  constructor(props: FavoriteBannerProps) {
+    super(props);
+    this.toggleClick = this.toggleClick.bind(this);
+  }
+
+  toggleClick(e: any) {
+    const { sharedId } = this.props;
+    const uwaziFavorites = getUwaziFavorites();
+
+    e.stopPropagation();
+    e.preventDefault();
+
+    if (uwaziFavorites.includes(sharedId)) {
+      const itemIndex = uwaziFavorites.indexOf(sharedId);
+      uwaziFavorites.splice(itemIndex, 1);
+    } else {
+      uwaziFavorites.push(sharedId);
+    }
+
+    localStorage.setItem('uwaziFavorites', uwaziFavorites.join(','));
+    this.forceUpdate();
+  }
+
+  render() {
+    if (isClient) {
+      const { sharedId } = this.props;
+      const selected = getUwaziFavorites().includes(sharedId);
+
+      return (
+        <button
+          className={`btn favoriteBanner ${selected ? 'selected' : ''}`}
+          onClick={this.toggleClick}
+          type="button"
+        >
+          <span className="tab-link-tooltip">Add / remove favorite</span>
+        </button>
+      );
+    }
+
+    return null;
```

Comment:
@RafaPolit does not this code produce a `Expected server HTML to contain a matching <div> in <div>` type of error, when reloading any library with already some favorites ?

Reply:
Not with 'some' favorites... if the collection has the feature enabled, it will show the warning.  I'll look into making this a bit better, but since the value is coming from localStorage, there is no way to fully avoid the issue of server rendering different than client.

Reply:
Actually, the React documentation is really good on this matter:
https://reactjs.org/docs/react-dom.html#hydrate

The 'isClient' implementation was completely avoided with a ComponentDidMount post-check.  This has the advantage of rendering in server the banners already if the functionality is enabled.  The downside is that it requires a re-render of all favored items afterwards, but that was already the case.

Nice catch!